src/lib.rs

```diff
@@ -699,6 +699,10 @@ pub trait FileExt {
 
     /// Update timestamps (both access and modification) to the current time.
     fn update_timestamps(&self) -> io::Result<()>;
+
+    /// Read the exact number of bytes required to fill `buf` starting from `position`,
+    /// without affecting file offset.
+    fn pread_exact(&self, buf: &mut [u8], position: usize) -> io::Result<()>;
```

Comment:
Hmm one thing I hadn't noticed before is how we were extending `FileExt`.  I had originally viewed that as a short term quick hack to just add the reflink-oriented copy (which itself I feel should land in `std`).  This repository is otherwise just methods extending the `openat` crate.  I think it'd be cleanest if these file helpers should be in a separate crate.  Some of them may already exist elsewhere.  But there's a *lot* of crates around file stuff too (https://lib.rs/filesystem). I don't know what's best.  Though one thing I do reference sometimes is dependencies of important bits in the ecosystem, e.g. `cargo`'s deps: https://crates.io/crates/cargo/0.54.0/dependencies 

From there I notice e.g. it depends on https://crates.io/crates/filetime/0.2.14 which may have the equivalent of `update_timestamps` already?

I think coreos-installer has some similar methods.

Anyways not a blocker, but let's keep this in mind?